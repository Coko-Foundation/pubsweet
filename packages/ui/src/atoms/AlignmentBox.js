import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

const lightGrey = '#b3b3b3'
const mainGrey = '#666'
const whiteRGBA = 'rgba(255, 255, 255, 1)'

const borderRule = position => props => `
    border-${position}-width: ${props.noBorder[position] ? '0' : '1px'}
  `

const Root = styled.div.attrs({
  role: 'presentation',
})`
  box-shadow: inset 0 0 0 2px ${whiteRGBA};
  cursor: pointer;
  height: 26px;
  width: 17px;

  border-style: solid;
  border-color: ${mainGrey};
  ${borderRule('top')};
  ${borderRule('right')};
  ${borderRule('bottom')};
  ${borderRule('left')};

  background-color: ${props => (props.active ? mainGrey : 'transparent')};

  &:hover {
    background-color: ${lightGrey};
  }
`

const AlignmentBox = ({ active, id, noBorder, onClick }) => (
  <Root active={active} id={id} noBorder={noBorder} onClick={onClick} />
)

AlignmentBox.propTypes = {
  active: PropTypes.bool.isRequired,
  id: PropTypes.string.isRequired,
  noBorder: PropTypes.shape({
    top: PropTypes.bool,
    bottom: PropTypes.bool,
    right: PropTypes.bool,
    left: PropTypes.bool,
  }),
  onClick: PropTypes.func,
}

AlignmentBox.defaultProps = {
  noBorder: {
    top: false,
    bottom: false,
    right: false,
    left: false,
  },
  onClick: () => null,
}

export default AlignmentBox
