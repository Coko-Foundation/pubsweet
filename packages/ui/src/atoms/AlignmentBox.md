Inactive

```js

const active = false;
const id = 'eyedee';
const noBorder = {};
const onClick = () => { return null };

<AlignmentBox
  active={active}
  id={id}
  noBorder={noBorder}
  onClick={onClick}
/>
```

Active

```js

const active = true;
const id = 'eyedee';
const noBorder = {};
const onClick = () => { return null };

<AlignmentBox
  active={active}
  id={id}
  noBorder={noBorder}
  onClick={onClick}
/>
```

Active with no right border

```js

const active = true;
const id = 'eyedee';
const noBorder = { right: true };
const onClick = () => { return null };

<AlignmentBox
  active={active}
  id={id}
  noBorder={noBorder}
  onClick={onClick}
/>
```
