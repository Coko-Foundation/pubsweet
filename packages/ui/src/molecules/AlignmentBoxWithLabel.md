AligmentBox with Label on the left

```js

const active = true;
const id = 'eyedee';
const labelText = 'left';
const noBorder = {};
const onClick = () => { return null };

<AlignmentBoxWithLabel
  active={active}
  id={id}
  labelText={labelText}
  noBorder={noBorder}
  onClick={onClick}
/>
```

Inactive AligmentBox with Label on the right

```js

const active = false;
const id = 'eyedee';
const labelPositionRight = true;
const labelText = 'right';
const noBorder = {};
const onClick = () => { return null };

<AlignmentBoxWithLabel
  active={active}
  id={id}
  labelPositionRight={labelPositionRight}
  labelText={labelText}
  noBorder={noBorder}
  onClick={onClick}
/>
```
