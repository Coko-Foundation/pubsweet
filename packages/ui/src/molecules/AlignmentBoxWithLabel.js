import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import fromTheme from '../helpers/fromTheme'

import AlignmentBox from '../atoms/AlignmentBox'

const blue = '#0d78f2'

const Root = styled.div`
  align-items: center;
  display: flex;
  flex-direction: row;
`

const Label = styled.span`
  color: ${blue};
  font-family: ${fromTheme('fontInterface')};
  font-size: 16px;
  font-style: italic;
  margin-left: ${props => (props.positionRight ? '10px' : '0')};
  margin-right: ${props => (props.positionRight ? '0' : '10px')};
  order: ${props => (props.positionRight ? '2' : '0')};
`

const AlignmentBoxWithLabel = ({
  active,
  id,
  labelPositionRight,
  labelText,
  noBorder,
  onClick,
}) => (
  <Root>
    <Label positionRight={labelPositionRight}>{labelText}</Label>
    <AlignmentBox
      active={active}
      id={id}
      noBorder={noBorder}
      onClick={onClick}
    />
  </Root>
)

AlignmentBoxWithLabel.propTypes = {
  active: PropTypes.bool.isRequired,
  id: PropTypes.string.isRequired,
  labelPositionRight: PropTypes.bool,
  labelText: PropTypes.string.isRequired,
  noBorder: PropTypes.shape({
    top: PropTypes.bool,
    bottom: PropTypes.bool,
    right: PropTypes.bool,
    left: PropTypes.bool,
  }),
  onClick: PropTypes.func,
}

AlignmentBoxWithLabel.defaultProps = {
  labelPositionRight: false,
  noBorder: {
    top: false,
    bottom: false,
    right: false,
    left: false,
  },
  onClick: () => null,
}

export default AlignmentBoxWithLabel
