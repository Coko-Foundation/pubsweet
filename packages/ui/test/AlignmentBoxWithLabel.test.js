import React from 'react'
import { shallow, mount, render } from 'enzyme'
import renderer from 'react-test-renderer'

import AlignmentBox from '../src/atoms/AlignmentBox'
import AlignmentBoxWithLabel from '../src/molecules/AlignmentBoxWithLabel'

const props = {
  active: true,
  id: 'left',
  labelPositionRight: true,
  labelText: 'Left',
}

const requiredProps = {
  active: false,
  id: 'left',
  labelText: 'Left',
}

const wrapper = shallow(<AlignmentBoxWithLabel {...requiredProps} />)
const wrapperMounted = mount(<AlignmentBoxWithLabel {...requiredProps} />)
const wrapperRendered = render(<AlignmentBoxWithLabel {...requiredProps} />)

describe('AlignmentBoxWithLabel', () => {
  test('is rendered correctly', () => {
    const tree = renderer
      .create(<AlignmentBoxWithLabel {...requiredProps} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should contain one Alignment Box child', () => {
    expect(wrapper.find(AlignmentBox).exists()).toEqual(true)
  })

  test('gets the correct props', () => {
    const noBorderDefaults = {
      top: false,
      bottom: false,
      right: false,
      left: false,
    }

    expect(wrapperMounted.prop('active')).toEqual(requiredProps.active)
    expect(wrapperMounted.prop('id')).toEqual(requiredProps.id)
    expect(wrapperMounted.prop('labelText')).toEqual(requiredProps.labelText)
    expect(wrapperMounted.prop('noBorder')).toEqual(noBorderDefaults)
    expect(wrapperMounted.prop('labelPositionRight')).toEqual(false)
  })

  test(`with default props the label is rendered on the left`, () => {
    expect(wrapper.is('.reverseOrder')).toBe(false)
    expect(wrapperRendered.text()).toEqual(requiredProps.labelText)
  })

  test('with given props has the correct classes', () => {
    const newWrapper = shallow(<AlignmentBoxWithLabel {...props} />)

    expect(newWrapper.is('.reverseOrder')).toBe(true)
  })
})
