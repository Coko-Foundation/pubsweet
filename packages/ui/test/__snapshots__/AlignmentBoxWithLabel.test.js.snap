// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlignmentBoxWithLabel is rendered correctly 1`] = `
.c2 {
  box-shadow: inset 0 0 0 2px rgba(255,255,255,1);
  cursor: pointer;
  height: 26px;
  width: 17px;
  border-style: solid;
  border-color: #666;
  border-top-width: 1px;
  border-right-width: 1px;
  border-bottom-width: 1px;
  border-left-width: 1px;
  background-color: transparent;
}

.c2:hover {
  background-color: #b3b3b3;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 {
  color: #0d78f2;
  font-size: 16px;
  font-style: italic;
  margin-left: 0;
  margin-right: 10px;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

<div
  className="c0"
>
  <span
    className="c1"
  >
    Left
  </span>
  <div
    className="c2"
    id="left"
    onClick={[Function]}
    role="presentation"
  />
</div>
`;
